bootstrap.sh

```diff
@@ -18,7 +18,7 @@ fi
 set -x
 
 if [[ "${RELEASE}" == "master" ]]; then
-  echo "NOTICE: the trace command is not available from master. RELEASE must be set to a released version (such as v0.2.0). See https://github.com/crossplaneio/crossplane-cli/releases for the full list of releases." >&2
+  { echo "NOTICE: the trace command is not available from master. RELEASE must be set to a released version (such as v0.2.0). See https://github.com/crossplaneio/crossplane-cli/releases for the full list of releases." ; } 2> /dev/null
```

Comment:
Wow, this is great! I didn't know about this syntax. The only issue I have is that the printed statement is now going to `stdout` instead of `stderr`. In general, results which can be used by another tool should go to `stdout`, and everything else (such as progress messages or warnings) should go to `stderr`, so our tool shouldn't send this message to `stdout`.

I just tried putting a redirect inside the braces and semicolon, but that seems to squelch the whole thing unfortunately. How about if we `set -x` after line 21 instead? So we'd need one `set -x` after line 21, and another `set -x` after line 29.

Reply:
Since the code is still executing, I wonder if the message should be deemed as an error or just information. Should't only fatal error messages go to `stderr`?

Reply:
> Should't only fatal error messages go to stderr?

No. The short version of the explanation is that `stdout` is what is used by the `|` operator, so results (and only results) should go into `stdout`, so they can be reused by another program by using the pipe operator. Everything else should go to `stderr`. "Everything else" usually means things such as logging and error messages, but it could be anything that isn't useful to another program later in the pipe chain.

There **are** reasons not to do it, but I typically like to assume that another (potentially unwritten) program will want to consume the output some day.

There's a bunch of reading that's available on this type of topic if you're interested! This is a subarea of [Unix Tool Philosophy](https://en.wikipedia.org/wiki/Unix_philosophy). There's a [fairly extensive book](http://www.catb.org/~esr/writings/taoup/html/) written by one of the people from the community about this and other related topics. Here's a [direct link to the section about pipes](http://www.catb.org/~esr/writings/taoup/html/ch07s02.html#plumbing) in case you're interested.

Reply:
@suskin  This seems to be a bit opinionated. One of the issues that I take here is that we have a message with a "Notice" prefix, and it's being deemed as error. I understand we don't want this message to appear in stdout for clear piping, but it also is not clear why could be piped out of this command.

All that said, I'm fine with the `set -x approach`, will implement that.